frontend/catalyst/ag_utils.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utility functions for Catalyst's AutoGraph module. This module can be safely imported without
+a TensorFlow installation."""
+
+import inspect
+
+import pennylane as qml
+
+# pylint: disable=import-outside-toplevel
+
+
+class AutoGraphError(Exception):
+    """Errors related to Catalyst's AutoGraph module."""
+
+
+def _test_ag_import():
+    """Reusable function for attempting to import Catalyst's AutoGraph module, which requires
+    TensorFlow to be installed."""
+
+    try:
+        import catalyst.autograph  # pylint: disable=unused-import
+    except ImportError as e:
+        raise ImportError(
+            "The AutoGraph feature in Catalyst requires TensorFlow. "
+            "Please install it (e.g. `pip install tensorflow-cpu`) and make it sure is "
+            "available in the current environment."
+        ) from e
+
+
+def converted_code(fn):
```

Comment:
Minor, but it could be nice to include autograph in the function name, since this is one of the main purposes of the function (and likely a term someone would use to search for this function).

I'm trying to think of something with one underscore only but can't think of anything 🤔 `autograph_converted_code` is too long.

Reply:
`autograph_source`?

Reply:
Yeah, we can go for `autograph_source` :)

Reply:
https://github.com/PennyLaneAI/catalyst/pull/235/commits/a3fce6505cf20a45fff47e922d1a849ea39aeb74